### ISSUE TITLE
WalletSignTransactionError: Something went wrong.

### ISSUE THREAD

**Issue by stanleycyang** [solana-labs/wallet-adapter#581]
**Describe the bug**

When I'm testing in local development (w/ mainnet-beta and custom RPC), our airdrop code works perfectly. but when moving it into prod, it complains `WalletSignTransactionError: Something went wrong.`. I checked all the ENV variables and made sure it was the same. 

**To Reproduce**

```
const transaction = new Transaction();

      // create token account
      if (receiverAccount == null) {
        transaction.add(
          createAssociatedTokenAccountInstruction(
            sender,
            receiverATA,
            receiver,
            mint
          )
        );
      }

      // add transfer instruction
      transaction.add(
        createTransferCheckedInstruction(
          senderATA,
          mint,
          receiverATA,
          sender,
          1,
          0
        )
      );

      const recentBlockhash = await connection.getLatestBlockhash();
      transaction.recentBlockhash = recentBlockhash?.blockhash;
      transaction.feePayer = sender;

      await signTransaction(transaction);

```

**Expected behavior**
I expected to work in both production and development

**Screenshots**
<img width="1371" alt="Screen Shot 2022-10-04 at 3 45 21 PM" src="https://user-images.githubusercontent.com/3037893/193944357-e6fc4cf1-da1f-4fc2-b30f-fc4368c22c3a.png">
<img width="363" alt="Screen Shot 2022-10-04 at 3 45 18 PM" src="https://user-images.githubusercontent.com/3037893/193944363-2df866e1-299d-4d7c-83a5-6a3ed3095766.png">

<img width="911" alt="Screen Shot 2022-10-04 at 5 13 04 PM" src="https://user-images.githubusercontent.com/3037893/193952931-cf7e6f3f-b0a9-44ce-924d-cff700967b68.png">

**Desktop (please complete the following information):**
 - OS: MacOS monterey
 - Browser chrome
 - Version latest

**Additional context**
NA

**Comment by jordansexton** [comment]
If the signing request is reaching the wallet, this doesn't seem related to Wallet Adapter. Unfortunately we can't help troubleshoot issues with wallets or deployment environments.

**Comment by jordansexton** [comment]
Also just noting, your reproduction code isn't something I can run. Lots of variables not defined, and you said it works in development against mainnet anyway. This points to an issue with your production build or environment.

**Comment by stanleycyang** [comment]
Fair enough. I am seeing the try catch fail specifically around the `signTransaction` in production though, which is odd. is there some potentially this package works differently in nextjs => vercel?

**Comment by stanleycyang** [comment]
I do see this as a recurring issue in the solana discord as well: 
<img width="548" alt="Screen Shot 2022-10-04 at 7 51 29 PM" src="https://user-images.githubusercontent.com/3037893/193971842-9109b2d9-4dd0-484c-a430-68528a32c1e9.png">

**Comment by jordansexton** [comment]
That's not a recurring issue. That's two different issues with different JS libraries under different circumstances. Their commonality is most likely user error.

Please follow up if you find a bug that can be reproduced with a complete repo. For example, if you provide a repo with a Next.js that works locally on mainnet but fails when deployed, I can take a look.

**Comment by stanleycyang** [comment]
Hey @jordansexton, here is a test repo: https://conversion-test.vercel.app/

Here is the code: https://github.com/fellowshipdao/conversion-test. I just ramped it up locally with some of the example repo code and it seems like it works locally but not in prod. any idea?

**Comment by jordansexton** [comment]
Thanks, looking into this!

**Comment by stanleycyang** [comment]
Hearing some talk about other people having issues on deployment to Vercel now as well with the core solana labs libs. I'm having trouble debugging since the error code is fairly generic. any ideas on what I did wrong here?

**Comment by jordansexton** [comment]
Can you try to deploy the [Next.js starter project](https://github.com/solana-labs/wallet-adapter/tree/master/packages/starter) and tell me if there are issues there? I've run this locally without a problem, but haven't deployed it recently. You may have to update the config to include the wallets you're testing.

**Comment by stanleycyang** [comment]
I believe I deployed almost that code in the other instance. I can try and clone and deploy this one

**Comment by kimpers** [comment]
For what it's worth this does not seem to have anything to do with Vercel but rather the Next.js build process. Running `yarn next build && yarn next start` will cause it to break when served locally as well

**Comment by jordansexton** [comment]
Thanks! I'll test that.

**Comment by vasudeveloper001** [comment]
This fails in local after build as well. Have similar issue but not entirely same, here in my devnet (so it doesn't have any relation with devnet, testnet, mainnet) -> [https://nextsol.ibcoder.com](https://nextsol.ibcoder.com )

It definitely fails during build in nextjs.
![image](https://user-images.githubusercontent.com/56103269/195992735-c452b308-027c-4e42-b3dd-52f9057a1986.png)

Using source-map debugging, it is pointing to wallet-adapter's code around here:

**adapter.js code piece pointed by source-maps.**
```js
async sendTransaction(transaction, connection, options = {}) {
        try {
            const wallet = this._wallet;
            if (!wallet)
                throw new WalletNotConnectedError();
            try {
                const { signers, ...sendOptions } = options;
                transaction = await this.prepareTransaction(transaction, connection, sendOptions);
                signers?.length && transaction.partialSign(...signers);
                sendOptions.preflightCommitment = sendOptions.preflightCommitment || connection.commitment;
                const { signature } = await wallet.signAndSendTransaction(transaction, sendOptions);
                return signature;
            }
            catch (error) {
                if (error instanceof WalletError)
                    throw error;
                throw new WalletSendTransactionError(error?.message, error);
            }
        }
        catch (error) {
            this.emit('error', error);
            throw error;
        }
    }
```
**Wallet-Provider calls adapter.js above**
```js
const sendTransaction = useCallback(
  async (transaction, connection, options) => {
    if (!adapter) throw handleError(new WalletNotSelectedError());
    if (!connected) throw handleError(new WalletNotConnectedError());
    return await adapter.sendTransaction(transaction, connection, options);
  },
  [adapter, handleError, connected]
);
```

Weird thing, this used to work like few months back!

**Comment by kimpers** [comment]
It seems like this issue is related to some change with the build/tree shaking process for Next.js 12. Downgrading to Next.js `^11.1.4` resolved the issue for me

**Comment by jordansexton** [comment]
Hmm, the example and nextjs-starter projects both use Next 12 and build and run locally for me. I'll try with them moved out of the workspace in case it's a dependency resolution issue.

**Comment by mahesh471978** [comment]
> Can you try to deploy the [Next.js starter project](https://github.com/solana-labs/wallet-adapter/tree/master/packages/starter) and tell me if there are issues there? I've run this locally without a problem, but haven't deployed it recently. You may have to update the config to include the wallets you're testing.

I had exactly same problem, as mentioned in top, and it got resolved when I used Next.js starter project by wallet-adapter.

**Comment by YoungMahesh** [comment]
> Can you try to deploy the [Next.js starter project](https://github.com/solana-labs/wallet-adapter/tree/master/packages/starter) and tell me if there are issues there? I've run this locally without a problem, but haven't deployed it recently. You may have to update the config to include the wallets you're testing.

@jordansexton , I had exactly same problem mentioned by @stanleycyang on the top and using Next.js starter **resolved** my issue. (I also tried `Downgrading to Next.js ^11.1.4` as mentioned by @kimpers but it didn't worked for me)

**Comment by jordansexton** [comment]
@YoungMahesh great! I might go ahead and do some testing with a vanilla Next app just to understand what the issue is and why the starter works if that doesn't.

**Comment by steveluscher** [comment]
I'm going to guess this is related to https://github.com/vercel/next.js/issues/41527 and fixed by https://github.com/swc-project/swc/pull/6201.

Were you folks using the `swcMinify: true` option?

**Comment by vasudeveloper001** [comment]
@steveluscher You're right. swcMinify is the problem. Setting it to false in next.config.js makes the site works: [https://nextsol.ibcoder.com](https://nextsol.ibcoder.com)

**Comment by steveluscher** [comment]
This is fixed in vercel/next v12.3.2-canary.32. Update and you're on your way!
